frontend/src/common.rs

```diff
@@ -884,7 +907,14 @@ pub fn plot_bar_chart(
         root.present()?;
     }
 
-    Ok(result)
+    Ok(Some(result))
+}
+
+fn all_zeros(data: &[(Vec<(NaiveDate, f32)>, usize)]) -> bool {
+    return data
+        .iter()
+        .map(|p| p.0.iter().map(|s| s.1 == 0.0).all(|e| e))
```

Comment:
Couldn't this be simplified to the following?

```suggestion
        .map(|p| p.0.iter().all(|s| s.1 == 0.0))
```

Reply:
The first `map` must be replaced by `all`, too. Then it's much more concise - I changed that.

Reply:
Yes, that's even better. :+1: